docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,130 @@
+# Faker connector
+
+The Faker connector generates realistically looking random data matching a
+specific schema.
+
+## Configuration
+
+To configure the Faker connector, create a catalog properties file
+`etc/catalog/example.properties` with the following contents:
+
+```text
+connector.name=faker
+```
+
+## Examples
+
+Faker generates data when reading from a table created in a catalog using this
+connector. This makes it easy to fill an existing schema with random data, by
+copying only the schema into a Faker catalog, and inserting the data back into
+the original tables.
+
+```sql
+CREATE TABLE faker.default.customer (LIKE production.public.customer EXCLUDING PROPERTIES);
+
+INSERT INTO production.public.customers
+SELECT *
+FROM faker.default.customers
+WHERE
+  born_at BETWEEN CURRENT_DATE - INTERVAL '150' YEAR AND CURRENT_DATE
+  AND age_years BETWEEN 0 AND 150
+LIMIT 100;
+```
+
+To generate more realistic data, choose specific generators by setting the
+`generator` property on columns:
+
+```sql
+SHOW CREATE TABLE production.public.customers;
+
+-- copy the output of the above query and add some properties:
+CREATE TABLE faker.default.customer (
+  id UUID NOT NULL,
+  name VARCHAR NOT NULL WITH (generator = '#{Name.first_name} #{Name.last_name}'),
+  address VARCHAR NOT NULL WITH (generator = '#{Address.fullAddress}'),
+  born_at DATE,
+  age_years INTEGER
+);
+```
+
+See the Datafaker's documentation for more information about
+[the expression](https://www.datafaker.net/documentation/expressions/) syntax
+and [available providers](https://www.datafaker.net/documentation/providers/).
+
+## Generators
```

Comment:
I kinda think we should document this first.. and then how to use the generator to insert data into somewhere else

also .. 

first how to create a plain generator table straight without any existing table 

then how to make it more complex and tweak it

then how to emulate another table

Reply:
There are two use cases for this connector:
1. Populating an existing database with realistically looking data, to test another application using this database.
2. Testing Trino itself, by populating some data source, and running queries that read data from it using another connector, like Iceberg. 

Only generating data, without copying an existing schema, is something you can do without this connector.

Choosing generators is optional and more complex than just doing a `CREATE TABLE LIKE`. 

We also don't recommend benchmarking Trino by reading from random data generators, as it's CPU bound, and doesn't utilize a lot of other parts of the engine.

I understand that the structure you proposed start with a simple task, and is supposed to be a complete reference, but it's too abstract.

Reply:
This is great .. we should explain all that in the docs.